packages/shr-fhir-export/lib/extensions.js

```diff
@@ -538,6 +552,51 @@ class ExtensionExporter {
     }
   }
 
+  inlineSubextension(subExt, extension, field, baseId, baseExt) {
+    for (const ext of subExt.snapshot.element) {
+      const snapExt = common.cloneJSON(ext);
+      // Slice off the extraneous "Extension" that will start the id
+      const splitPoint = snapExt.id.indexOf('.');
+      const endId = snapExt.id.slice(splitPoint + 1);
+      // Construct a newId and path for the inlined version of the extension
+      const newId = `${baseId}.extension:${common.shortID(field.effectiveIdentifier)}.${endId}`;
+      const pathEnd = endId.slice(endId.lastIndexOf('.') + 1);
+      const existingElement = extension.snapshot.element.find(el => el.id === newId);
+      // Applying constraints can add elements that we don't actually want, just keep values
+      if (!pathEnd.startsWith('value')) {
+        extension.snapshot.element = extension.snapshot.element.filter(el => el.id !== newId);
+      }
+      // This avoids inlining the uneccessary top level "Extension" element or readding value that was constrained
+      if (splitPoint < 0 || (existingElement != null && pathEnd.startsWith('value'))) {
+        continue;
+      } else {
+        // Set the inlined version of the snapshot element with the path and id
+        snapExt.id = newId;
+        snapExt.path = this.getExtensionPathFromExtensionID(snapExt.id);
+        if (pathEnd === 'url') {
+          // Fix the "url" field to the name of the inlined extension, following example of USCore
+          snapExt.fixedUri = newId.slice(newId.lastIndexOf(':') + 1, newId.lastIndexOf('.'));
+        }
+      }
```

Comment:
Since the only thing that happens when your condition is true is that the loop restarts, it is possible to move the contents of the `else` block to just after the `if` block, and then remove the `else`. Alternatively, if you negate the condition, you can avoid needing to use a `continue` statement by placing the rest of the loop's contents in the `if` block.